testing/test_runner.py

```diff
@@ -570,6 +570,15 @@ def pytest_configure(config):
     result.stderr.fnmatch_lines(["Exit: oh noes"])
 
 
+def test_pytest_exit_returncode():
```

Comment:
Following my suggestion above, this will need to be changed to a test using `testdir`, something like:

```python
def test_pytest_exit_returncode(testdir):
    testdir.makepyfile("""
        import pytest
        def test_foo():
            pytest.exit("some exit msg", 99)
    """)
    result = testdir.runpytest()
    assert result.ret == 99
```